micronaut/src/main/java/software/amazonaws/example/product/PrimingResource.java

```diff
@@ -0,0 +1,34 @@
+package software.amazonaws.example.product;
+
+import io.micronaut.crac.OrderedResource;
+import jakarta.inject.Singleton;
+import org.crac.Context;
+import org.crac.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazonaws.example.product.dao.DynamoProductDao;
+
+@Singleton
+public class PrimingResource implements OrderedResource {
+  private static final Logger log = LoggerFactory.getLogger(PrimingResource.class);
+  private final DynamoProductDao dynamoProductDao;
+
+  public PrimingResource(DynamoProductDao dynamoProductDao) {
+    this.dynamoProductDao = dynamoProductDao;
+  }
+
+  public void beforeCheckpoint(Context<? extends Resource> context) throws Exception {
```

Comment:
Do you need to override the beforeCheckpoint() and afterCheckpoint() methods? Similar to Spring Boot pattern and in this Micronaut [blog](https://micronaut.io/2022/11/28/leveraging-aws-lambda-snapstart-with-the-micronaut-framework/).

Reply:
`micronaut-crac` does that under the hood.